src/scenic/simulators/carla/model.scenic

```diff
@@ -121,7 +130,8 @@ class Pedestrian(Pedestrian, CarlaActor, Walks):
     blueprint: Uniform(*blueprints.walkerModels)
 
     def setWalkingDirection(self, heading):
-        direction = Vector(0, self.speed).rotatedBy(heading)
+        forward = self.carlaActor.get_transform().get_forward_vector()
```

Comment:
This doesn't look right - the transform of the CARLA actor is irrelevant to what direction the pedestrian should walk (which is being given by `heading` in Scenic's coordinate system), and you're ignoring `self.speed`. What was wrong with the old calculation?